code/game/objects/items/weapons/pan.dm

```diff
@@ -0,0 +1,460 @@
+	if(istype(attackee, /mob/living/simple_animal/hostile/necro))
+		power *= 2 //L4D4EVR
+	return power
+
+/obj/item/weapon/reagent_containers/pan/empty_contents()
+	set name = "Dump contents"
+	set category = "Object"
+	set src in usr
+
+	if(usr.incapacitated())
+		to_chat(usr, "<span class='warning'>You can't do that while incapacitated.</span>")
+		return
+	if(!is_open_container(src))
+		return
+	if(!src.contains_anything())
+		to_chat(usr, "<span class='warning'>\The [src] is empty.</span>")
+		return
+	if(isturf(usr.loc))
+		usr.investigation_log(I_CHEMS, "has emptied \a [src] ([type]) containing [reagents.get_reagent_ids(1)] onto \the [usr.loc].")
+		drop_ingredients(usr.loc, usr)
+
+/////////////////////Cooking-related stuff/////////////////////
+
+/obj/item/weapon/reagent_containers/pan/proc/cook_start() //called when the pan is placed on a valid cooktop (eg. placed on grill)
+	var/contains_anything = contains_anything()
+	if(contains_anything)
+		fast_objects.Add(src)
+	if(contains_anything & COOKVESSEL_CONTAINS_CONTENTS)
+		currentrecipe = select_recipe(available_recipes,src)
+
+/obj/item/weapon/reagent_containers/pan/proc/cook_stop() //called when the pan is no longer on a valid cooktop (eg. removed from grill). cooking progress is retained unless things are added or removed from the pan
+	burned = FALSE
+	fast_objects.Remove(src)
+
+/obj/item/weapon/reagent_containers/pan/proc/cook_abort() //called when things are dumped out of the pan
+	cook_stop()
+	cook_reboot()
+
+/obj/item/weapon/reagent_containers/pan/proc/cook_reboot(mob/user) //called when we want to restart the cooking process eg. when something was added to the pan
+	reset_cooking_progress()
+	chef = user
+	currentrecipe = select_recipe(available_recipes,src)
+
+/obj/item/weapon/reagent_containers/pan/proc/cook_fail() //called when the recipe is invalid (including overcooking something by not removing the pan from the cooktop in time, which is the same as attempting to cook the already-cooked resulting dish)
+	var/obj/item/weapon/reagent_containers/food/snacks/badrecipe/ffuu = new(src)
+	var/amount = 0
+	for(var/obj/O in contents-ffuu)
+		amount++
+		if (O.reagents)
+			var/id = O.reagents.get_master_reagent_id()
+			if (id)
+				amount+=O.reagents.get_reagent_amount(id)
+		qdel(O)
+	reagents.clear_reagents()
+	ffuu.reagents.add_reagent(CARBON, amount)
+	ffuu.reagents.add_reagent(TOXIN, amount/10)
+	if(Holiday == APRIL_FOOLS_DAY)
+		playsound(src, "goon/sound/effects/dramatic.ogg", 100, 0)
+	ffuu.pixel_x = 0
+	ffuu.pixel_y = 0
+	currentrecipe = select_recipe(available_recipes, src)
+	burned = TRUE
+	return ffuu
+
+/obj/item/weapon/reagent_containers/pan/process()
+
+	var/obj/O
+	if(isobj(loc))
+		O = loc
+		if(!O.is_cooktop)
+			cook_stop()
+			return
+	else
+		cook_stop()
+		return
+
+	if(!(O?.can_cook())) //if eg. the power went out on the grill, don't cook
+		return
+
+	cookingprogress += (SS_WAIT_FAST_OBJECTS * speed_multiplier)
```

Comment:
it may make sense to render a progress bar for fry pan cooking to give people an idea of how far they are in the 10 seconds

Reply:
There's a visible_message and a sound that plays when it finishes, I kind of like the idea of needing to closely watch the pan or otherwise acquire the finesse of internalizing the timings to avoid it burning.